### PR TITLE
Improve performance of annotation db creation, querying

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: tests/test_core/test_annotation_db.py [hunk 1]
```diff
@@ -1207,3 +1222,64 @@ def test_subset_gff3_db_source(gff_db, tmp_dir):
...
+
+@pytest.mark.parametrize("db", ("gff_db", "gb_db"))
+@pytest.mark.parametrize(
+    "col", ("biotype", "seqid", "name", "start", "stop", "parent_id")
+)
+def test_db_make_index(request, db, col):
+    ann_db = request.getfixturevalue(db)
+    expect = {("index", col, tn) for tn in ann_db.table_names}
+    ann_db.make_indexes()
+    sql_template = f"SELECT * FROM sqlite_master WHERE type = 'index' AND tbl_name = '%s' and name = {col!r}"
```
Comment (khiron) [inline tests/test_core/test_annotation_db.py:1274]: You might be able to ignore the codacy sql injection warning with 

`# codacy:ignore[sql-injection] - limited SQL injection exposure`

As the only exposure I can see here is to unrestricted browsing, and it's just a test

Comment (GavinHuttley) [reply tests/test_core/test_annotation_db.py:1274]: Good point, will apply that.

File: src/cogent3/parse/gff.py [hunk 2]
```diff
@@ -1,21 +1,166 @@
...
+OptionalStr = typing.Optional[str]
+OptionalInt = typing.Optional[str]
+OptionalStrDict = typing.Optional[typing.Union[str, dict[str, str]]]
+OptionalBool = typing.Optional[bool]
+
+
+@functools.singledispatch
+def is_gff3(f) -> bool:
+    """True if gff-version is 3"""
+    raise TypeError(f"unsopported type type {type(f)}")
```
Comment (khiron) [inline src/cogent3/parse/gff.py:25]: uns_u_pported

**Review by khiron — APPROVED** [review]
one spelling error needs fixing other than that it's good

File: src/cogent3/parse/gff.py (hunk 2 above)
Comment (GavinHuttley) [reply src/cogent3/parse/gff.py:25]: and "type" x 2! Jeez

**Pull request merged**